core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -673,10 +673,21 @@ func (r *EvmRegistry) checkUpkeeps(ctx context.Context, payloads []ocr2keepers.U
 		checkResults = make([]*string, len(payloads))
 		blocks       = make([]*big.Int, len(payloads))
 		upkeepIds    = make([]*big.Int, len(payloads))
+		results      = make([]ocr2keepers.CheckResult, len(payloads))
 	)
 
 	for i, p := range payloads {
 		block, upkeepId := r.getBlockAndUpkeepId(p.UpkeepID, p.Trigger)
+		if r.lastPollBlock-block.Int64() > maxLookbackDepth {
```

Comment:
when is `r.lastPollBlock` updated? (From the name it doesn't seem like it would be the latest block)

Reply:
alternatively I'm not sure if we have easy access to block subscriber here which should be able to give latest block

Reply:
also, can we add a test for this behaviour